### PR TITLE
Verify that dynamic filters in plan are above table scan

### PR DESCRIPTION
**Pull request by raunaqmorarka**

### CODE REVIEW

File: presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestRemoveUnsupportedDynamicFilters.java [hunk 1]
```diff
@@ -182,6 +182,39 @@ public void testUnmatchedDynamicFilter()
...
+                        Optional.empty(),
+                        Optional.empty(),
+                        ImmutableMap.of("DF", ordersOrderKeySymbol)));
+        assertPlan(
+                removeUnsupportedDynamicFilters(root),
+                output(
+                        join(
+                                INNER,
+                                ImmutableList.of(equiJoinClause("LINEITEM_OK", "ORDERS_OK")),
+                                filter("LINEITEM_OK > 0",
```
Comment (sopel39) [inline presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestRemoveUnsupportedDynamicFilters.java:213]: nit: each argument in newline

Comment (sopel39) [reply presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestRemoveUnsupportedDynamicFilters.java:213]: You should use `io.prestosql.sql.planner.assertions.PlanMatchPattern#filter(io.prestosql.sql.tree.Expression, io.prestosql.sql.tree.Expression, io.prestosql.sql.planner.assertions.PlanMatchPattern)` instead. This `filter` method doesn't validate dynamic filters.

Comment (raunaqmorarka) [reply presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestRemoveUnsupportedDynamicFilters.java]: Done

File: presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestRemoveUnsupportedDynamicFilters.java [hunk 2]
```diff
@@ -182,6 +182,41 @@ public void testUnmatchedDynamicFilter()
...
+                        Optional.empty(),
+                        ImmutableMap.of("DF", ordersOrderKeySymbol)));
+        assertPlan(
+                removeUnsupportedDynamicFilters(root),
+                output(
+                        join(
+                                INNER,
+                                ImmutableList.of(equiJoinClause("LINEITEM_OK", "ORDERS_OK")),
+                                filter(
+                                        PlanBuilder.expression("LINEITEM_OK > 0"),
```
Comment (sopel39) [inline presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestRemoveUnsupportedDynamicFilters.java:214]: static import `expression`

File: presto-main/src/test/java/io/prestosql/sql/planner/sanity/TestDynamicFiltersChecker.java [hunk 3]
```diff
@@ -140,6 +140,31 @@ public void testUnmatchedDynamicFilter()
...
+    {
+        PlanNode root = builder.output(
+                ImmutableList.of(),
+                ImmutableList.of(),
+                builder.join(
+                        INNER,
+                        builder.filter(
+                                combineConjuncts(
+                                        metadata,
+                                        PlanBuilder.expression("LINEITEM_OK > 0"),
```
Comment (sopel39) [inline presto-main/src/test/java/io/prestosql/sql/planner/sanity/TestDynamicFiltersChecker.java:154]: static import `expression`

File: presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestRemoveUnsupportedDynamicFilters.java (hunk 2 above)
Comment (raunaqmorarka) [reply presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestRemoveUnsupportedDynamicFilters.java]: done

File: presto-main/src/test/java/io/prestosql/sql/planner/sanity/TestDynamicFiltersChecker.java (hunk 3 above)
Comment (raunaqmorarka) [reply presto-main/src/test/java/io/prestosql/sql/planner/sanity/TestDynamicFiltersChecker.java]: done

**Pull request merged**